dbms/src/IO/WriteBufferFromFileDescriptor.cpp

```diff
@@ -51,6 +51,11 @@ void WriteBufferFromFileDescriptor::nextImpl()
 
         if ((-1 == res || 0 == res) && errno != EINTR)
         {
+            if (errno == EPIPE)
+            {
+                break;
```

Comment:
Almost Ok, but I'm not sure about this modification.
I think, `WriteBufferFromFileDescriptor` should report about broken pipe with exception.

Example 1: cannot write to pipe when `clickhouse-client` run in non-interactive mode:

```
clickhouse-client --query="SELECT number FROM system.numbers" | head -n10
```

Example 2: connection loss while writing to network in clickhouse-server.

I will check without this change.

Reply:
With this modification, it allows to silently write and ignore data after broken pipe.
It will consume time and CPU in some cases.

Reply:
I tested without this modification:

```
milovidov@milovidov-Pro-P30:~/work/ClickHouse/dbms/src/Server$ ./clickhouse client --pager=head
ClickHouse client version 1.1.54326.
Connecting to localhost:9000.
Connected to ClickHouse server version 1.1.54326.

:) SELECT number FROM system.numbers

SELECT number
FROM system.numbers 

┌─number─┐
│      0 │
│      1 │
│      2 │
│      3 │
│      4 │
│      5 │
│      6 │
│      7 │
│      8 │

Exception on client:
Code: 75. DB::Exception: Cannot write to file (fd = 7), errno: 32, strerror: Broken pipe

Connecting to localhost:9000.
Connected to ClickHouse server version 1.1.54326.

:) Bye.
```

Looks totally fine except reconnecting.

Reply:
It passes your test (though I had to install `expect`, but that's ok).